File: src/main/java/com/hascode/tutorial/service/BookBean.java
package com.hascode.tutorial.service;

import java.util.UUID;

import com.hascode.tutorial.entity.Book;

public class BookBean {
	public Book create(final String title) {
		if (title == null) {
			throw new IllegalArgumentException("title must be set");
		}
		return new Book(title, UUID.randomUUID().toString().toUpperCase());
	}
}

File: src/main/scala/com/hascode/tutorial/service/BookBean.scala
package com.hascode.tutorial.service

import java.util.UUID

import com.hascode.tutorial.entity.Book

object BookBean {private val MIN_LENGTH: Int = 3}

class BookBean {
  def create(title: String) = {
    if (title == null) {
      throw new IllegalArgumentException("title must be set")
    }
    if (title.length <= BookBean.MIN_LENGTH) {
      throw new IllegalArgumentException("title must have a minimal length of " + BookBean.MIN_LENGTH)
    }
    new Book(title, UUID.randomUUID.toString.toUpperCase)
  }
}